### PR TITLE
Implemented the given user stories

### PR DESCRIPTION
**Pull request by Hives**
Outstanding:

- Edge cases
- Refactor?
- Code quality review
- $BONUS task

### CODE REVIEW

File: spec/airport_spec.rb [hunk 1]
```diff
@@ -0,0 +1,230 @@
+require 'airport'
+
+describe Airport do
```
Comment (samanthaixer) [inline spec/airport_spec.rb]: Good set of tests that seem to cover a range of scenarios. Perhaps because of the volume of tests, I am sometimes finding it hard to follow what is being tested at each point. I will highlight at least one example below

File: spec/airport_spec.rb [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+      end
+
+      context 'when the plane is already at the airport' do
+        before :each do
+          allow(@weather).to receive(:stormy?).and_return false 
+          @airport.land @plane
+          allow(@weather).to receive(:stormy?).and_return true
+        end
+
+        it 'raises an error' do
```
Comment (samanthaixer) [inline spec/airport_spec.rb]: I have found this set of tests a little confusing, perhaps just the ordering of the stubs and context descriptions

File: README.md [hunk 3]
```diff
@@ -1,90 +1,52 @@
...
-2. Run the command `gem install bundle` (if you don't have bundle already)
-3. When the installation completes, run `bundle`
-4. Complete the following task:
-
-Task
------
+## Instructions
 
-We have a request from a client to write the software to control the flow of planes at an airport. The planes can land and take off provided that the weather is sunny. Occasionally it may be stormy, in which case no planes can land or take off.  Here are the user stories that we worked out in collaboration with the client:
+In root folder execute `irb -r ./lib/airport.rb -r ./lib/weather_station.rb -r ./lib/plane.rb`.
```
Comment (mattstockman) [inline README.md]: Clear instructions on how to run the programme

File: README.md [hunk 4]
```diff
@@ -1,90 +1,52 @@
...
+After implementing the features from the user stories I remembered about RSpec `context` blocks and spent a while refactoring all my tests...
 
-Reviewers will potentially be using this [code review rubric](docs/review.md).  Referring to this rubric in advance will make the challenge somewhat easier.  You should be the judge of how much challenge you want this weekend.
+After that I started writing user stories for the edge cases, and using the same process to implement them. I ran out of time before finishing all of the edge cases, but here's how I would have continued:
 
-**BONUS**
+### Outstanding edge cases
 
-* Write an RSpec **feature** test that lands and takes off a number of planes
+- Planes that are flying cannot take off / planes that are not flying cannot land
```
Comment (mattstockman) [inline README.md]: all steps clearly outlined

File: lib/airport.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
+class Airport
+  DEFAULT_CAPACITY = 3
+
+  def initialize(weather, capacity = DEFAULT_CAPACITY)
```
Comment (samanthaixer) [inline lib/airport.rb]: Could we discuss the benefits to setting the default in the initialize arguments versus as a constant outside the initialize?

File: lib/airport.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  end
+  
+  def land(plane)
+    raise "That plane is already on the ground!" if @planes.include? plane
+    raise "⛈ Too dangerous to land. ⛈" if bad_weather?
+    raise "Airport full" if @planes.size >= @capacity
+
+    @planes << plane
+    puts '☀️ Plane landed! ☀️'   
+    self
```
Comment (samanthaixer) [inline lib/airport.rb]: What is self here? And is there a reason that the confirmation message is output to the screen rather than returned by the method?

File: lib/airport.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  end
+
+  def include?(plane)
+    @planes.include? plane
+  end
+
+  private
+
+  def bad_weather?
```
Comment (samanthaixer) [inline lib/airport.rb]: Good naming convention for method. It makes reading the code where this method is called very easy

File: lib/airport.rb [hunk 8]
```diff
@@ -0,0 +1,42 @@
+class Airport
```
Comment (mattstockman) [inline lib/airport.rb]: nicely indented making it easy to read

File: lib/plane.rb [hunk 9]
```diff
@@ -0,0 +1,21 @@
+class Plane
+  def initialize
+    @flying = true
+  end
+  
+  def flying?
```
Comment (samanthaixer) [inline lib/plane.rb]: Could this be an attribute reader instead of method?

File: lib/plane.rb [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+    @flying
+  end
+
+  def land
```
Comment (samanthaixer) [inline lib/plane.rb]: Good simple methods that are easy to understand

File: lib/plane.rb (hunk 9 above)
Comment (Hives) [reply lib/plane.rb]: yes!

File: spec/airport_spec.rb (hunk 2 above)
Comment (Hives) [reply spec/airport_spec.rb]: yes me too

File: lib/airport.rb (hunk 5 above)
Comment (Hives) [reply lib/airport.rb]: done

File: lib/airport.rb (hunk 6 above)
Comment (Hives) [reply lib/airport.rb]: not sure if i've done the right thing here, i will raise at the code review

File: lib/airport.rb (hunk 7 above)
Comment (Hives) [reply lib/airport.rb]: ta!
